src/components/DailyTasks.js

```diff
@@ -1,48 +1,84 @@
-import {useEffect, useState} from 'react';
+import {useState, useEffect} from 'react';
+import {dailyTodos} from '../db';
 import {loadFromLocalStorage, writeToLocalStorage} from '../util/localstorage';
-import styled from 'styled-components';
 import TaskList from './TaskList';
-import dailyTasks from '../db';
 
 export default function DailyTasks() {
+  const [daily, monday, tuesday, wednesday, thursday, friday, weekend] = dailyTodos;
+  let currentday;
+  switch (new Date().getDay()) {
+    case 0:
+      currentday = [weekend];
+      break;
+    case 1:
+      currentday = [daily, monday];
+      break;
+    case 2:
+      currentday = [daily, tuesday];
+      break;
+    case 3:
+      currentday = [daily, wednesday];
+      break;
+    case 4:
+      currentday = [daily, thursday];
+      break;
+    case 5:
+      currentday = [daily, friday];
+      break;
+    case 6:
+      currentday = [weekend];
+      break;
+    default:
+  }
+
   const [todos, setTodos] = useState(() => {
     const todosFromLocal = loadFromLocalStorage('todos');
-    const uncheckedTodos = todosFromLocal?.map(todo =>
-      todo.checkedAt !== new Date().toLocaleDateString() ? {...todo, isChecked: false, checkedAt: ''} : todo
-    );
-    return uncheckedTodos ?? dailyTasks;
+    const uncheckedTodos = todosFromLocal?.map(todo => {
+      if (todo.tasks !== undefined) {
+        return {
+          ...todo,
+          tasks: todo.tasks.map(task =>
+            task.checkedAt !== new Date().toLocaleDateString() ? {...task, isChecked: false, checkedAt: ''} : task
+          ),
+        };
+      }
+      return todo;
+    });
+    return uncheckedTodos ?? currentday;
```

Comment:
Since you have quite a bit of logic here, you might want to consider creating a custom hook and extract the logic :)